src/encoding.rs

```diff
@@ -568,6 +568,16 @@ impl EncodeGaugeValue for i64 {
     }
 }
 
+impl EncodeGaugeValue for u64 {
+    fn encode(&self, encoder: &mut GaugeValueEncoder) -> Result<(), std::fmt::Error> {
+        if *self == u64::MAX {
+            return Err(std::fmt::Error);
+        }
```

Comment:
Can you add a comment here, summarizing your PR description. Makes it is easier for me in the future without going through the git history.